CONTRIBUTING.md

```diff
@@ -61,6 +61,49 @@ Pull requests should not be squash merged in order to keep complete commit messa
 This means that, when addressing review comments in order to keep the pull request in an always mergeable status, you will sometimes need to rewrite your branch's history and then force-push it with `git push --force-with-lease`.
 Useful git commands that can help a lot with this are `git commit --patch --amend` and `git rebase --interactive`. For more details consult the git man pages or online resources like [git-rebase.io](https://git-rebase.io/) or [The Pro Git Book](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History).
 
+## Including patches
+
+While we do accept custom patches, we strongly encourage submitting patches
+upstream whenever possible. This practice means submitting the patches against
+the original software's repository, which has the potential to benefit the
+broader user community.
+
+For out-of-tree patches fetched using `fetchpatch`, avoid linking to GitHub pull
+requests, as these are mutable and can change with each new commit or
+force-push. Instead, use direct links to specific commit patches such as
+https://github.com/owner/repo/commit/sha.patch. This ensures that the reference
+to your patch remains constant and accurate.
```

Comment:
I agree this ensure the reference remains constant, but we loose the context of the PR. A compromise is
https://github.com/owner/repo/pull/number/commits/sha.patch, which gets redirected to the version you propose

Reply:
Good suggestion. Fixed.